streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/Serdes.scala

```diff
@@ -41,8 +43,19 @@ object Serdes {
   implicit def Integer: Serde[Int] = JSerdes.Integer().asInstanceOf[Serde[Int]]
   implicit def JavaInteger: Serde[java.lang.Integer] = JSerdes.Integer()
 
-  implicit def timeWindowedSerde[T](implicit tSerde: Serde[T]): WindowedSerdes.TimeWindowedSerde[T] =
-    new WindowedSerdes.TimeWindowedSerde[T](tSerde)
+  implicit def timeWindowedSerde[T](implicit inner: Serde[T]): Serde[Windowed[T]] =
+    new JSerdes.WrapperSerde[Windowed[T]](
+      new TimeWindowedSerializer[T](inner.serializer),
+      new TimeWindowedDeserializer[T](inner.deserializer) {
+        override def deserialize(topic: String, data: Array[Byte]): Windowed[T] = {
```

Comment:
Thinking about this once again: if users did used the `timeWindowedSerde`, then we would log a WARN on each record, which would soon flood the log files. So instead of log at the `deserialize` function, maybe it's better to just log once at the `timeWindowedSerde` itself?

Reply:
Yeah, I agree that we don't want to spam the logs. There is the `deprecated` annotation above as well - with that do we even need the log messaged for the `timeWindowedSerde`?

Reply:
Sorry if I'm missing the point here, @lct45 , but why do we need any custom code at all in the scala wrapper? Shouldn't we just wrap the Java Serde, which you've already set up to warn upon `configure` if there's no constructor arg and no configured value?

Reply:
Hmm, maybe we don't need custom code? I did this implementation back in the fall before [KIP-616](https://cwiki.apache.org/confluence/display/KAFKA/KIP-616%3A+Rename+implicit+Serdes+instances+in+kafka-streams-scala) was merged. You know the scala better than I do, but I think you're right that the deprecated warning in the `TimeWindowedDeserializer` will be enough now that KIP-616 added the `deprecated` annotation to this class